src/main/resources/hudson/plugins/gradle/injection/init-script.gradle

```diff
@@ -137,12 +139,12 @@ if (GradleVersion.current() < GradleVersion.version('6.0')) {
                             logger.lifecycle("Enforcing Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer, captureTaskInputFiles: $geCaptureTaskInputFiles")
                             buildScan.server = geUrl
                             buildScan.allowUntrustedServer = geAllowUntrustedServer
-                            if (isAtLeast(gePluginVersion, '2.1')) {
-                                if (isNotAtLeast(gePluginVersion, '3.7')) {
-                                    buildScan.captureTaskInputFiles = geCaptureTaskInputFiles
-                                } else {
-                                    buildScan.capture.taskInputFiles = geCaptureTaskInputFiles
-                                }
+                        }
+                        if (geCaptureTaskInputFiles && isAtLeast(gePluginVersion, '2.1') && atLeastGradle5) {
```

Comment:
Is this block executed if https://github.com/jenkinsci/gradle-plugin/pull/408/files#diff-9026492162e503f2dfa14b8cb79801da9f4fd7425809e813bb2413b4f728bc0bL116-L132 is ? ie do we need to have it duplicated if we want to set it in all cases?

Reply:
Not necessarily; one block is executed if the plugin is not already applied and the other one is when enforceUrl is required (which happens for both cases where plugin is or isn't already applied).

Reply:
The logic is now extracted